### ISSUE TITLE
Attaching key file to a database afterwords not working

### ISSUE THREAD

**Issue by Juk111** [PhilippC/keepass2android#163]
Its very hard or even impossible to attach key files to a already made database. There is opportunkty to swich master password and there is also place for adding keyfile. No idea how!!? Fist of all no one can figure out the right directory path on android devices and i f you have old Keepassdroid you can check out the right path and copy paste the path. This should work? Does not. Even with the right file path it does not work. This is very unfortunent thing on otherwice so very usefull and handy app. Does it matter that these databases are originally made with Keepassdroid? They do work otherwice well with Keepass2Android. I have tested these functions with both online version and with offline version of the app.  I have been testing with different android devices with android 6.0. Key files can only be attached when creating a new database and then it is easy to do with the browsing function. Why isn't there  browsing function on the "change master password funtion also? Also if this is not possible it would be also handy to be able to move files from one database to another, possibly with one with key file. Key file is important if you wanna safe database online so it would be great to have this working smoothly.

**Comment by PhilippC** [comment]
I agree that this should be improved. But keep in mind that you also need a way to generate a key file then. And make users backup the keyfile... I suggest doing all this on the PC...

**Comment by jmichael2497** [comment]
> I suggest doing all this on the PC...

that's what used to be said about password managers in general ;) but nowadays, consider that globablly, not everyone has regular access to a computer of their own, maybe just an old mobile device.

since there is a lot of shared code between KeePassDroid, KeePass2Android, and KeePassDX then you may want to peek at KeePassDX for some code fixes and ideas (you two seem to have the most advanced cherry picks of features).

adding a file picker at that change master password (and new db) screen, made it possible to complete the task of making a password based db, and then switching it to pw+key db, in KeePassDX.

(which reminds me, seems weird that all 3 apps don't offer the ability to choose encryption scheme options when making the db, forced to change it from AES after creation)

nice work though, it's almost a coin toss between this offline version and DX.
